apps/studio/components/interfaces/Settings/General/ComplianceConfig/ProjectComplianceMode.tsx

```diff
@@ -0,0 +1,126 @@
+import { useCheckPermissions } from 'hooks/misc/useCheckPermissions'
+import { subscriptionHasHipaaAddon } from 'components/interfaces/Billing/Subscription/Subscription.utils'
+import { useOrgSubscriptionQuery } from 'data/subscriptions/org-subscription-query'
+import { useSelectedOrganization } from 'hooks/misc/useSelectedOrganization'
+
+import { Switch, TooltipContent_Shadcn_, TooltipTrigger_Shadcn_, Tooltip_Shadcn_ } from 'ui'
+
+const ComplianceConfig = () => {
+  const { ref } = useParams()
+  const [isSensitive, setIsSensitive] = useState(false)
+  const selectedOrganization = useSelectedOrganization()
+  const { data: subscription } = useOrgSubscriptionQuery({ orgSlug: selectedOrganization?.slug })
+  const hasHipaaAddon = subscriptionHasHipaaAddon(subscription)
+  const {
+    data: settings,
+    isLoading,
+    isSuccess,
+  } = useProjectSettingsV2Query({ projectRef: ref }, { enabled: hasHipaaAddon })
+
+  const { mutate: updateComplianceConfig, isLoading: isSubmitting } =
+    useComplianceConfigUpdateMutation({
+      onSuccess: () => {
+        toast.success('Successfully updated project compliance configuration')
+      },
+      onError: (error) => {
+        setIsSensitive(initialIsSensitive)
+        toast.error(`Failed to update project compliance configuration: ${error.message}`)
+      },
+    })
+
+  const { project } = useProjectContext()
+  const canUpdateComplianceConfig = useCheckPermissions(PermissionAction.UPDATE, 'projects', {
+    resource: {
+      project_id: project?.id,
+    },
+  })
+  const initialIsSensitive = settings?.is_sensitive || false
+
+  useEffect(() => {
+    if (!isLoading) {
+      setIsSensitive(initialIsSensitive)
+    }
+  }, [isLoading])
+
+  const toggleIsSensitive = async () => {
+    if (!ref) return console.error('Project ref is required')
+    setIsSensitive(!isSensitive)
+    updateComplianceConfig({ projectRef: ref, isSensitive: !isSensitive })
+  }
+
+  // this is only setable on compliance orgs, currently that means HIPAA orgs
+  if (!hasHipaaAddon) {
+    return
+  }
+
+  return (
+    <div id="compliance-configuration">
+      <div className="flex items-center justify-between mb-6">
+        <FormHeader className="mb-0" title="High Compliance Configuration" description="" />
+        <DocsButton href="https://supabase.com/docs/guides/deployment/shared-responsibility-model#managing-healthcare-data" />
+      </div>
+      <FormPanel>
+        <FormSection
+          header={
+            <FormSectionLabel
+              className="lg:col-span-7"
+              description={
+                <div className="space-y-4">
+                  <p className="text-sm text-foreground-light">
+                    Apply additional compliance controls to this project
+                  </p>
+                </div>
+              }
+            >
+              Project processes sensitive data (HIPAA)
```

Comment:
I don't quite understand what this label means - ideally the language should be actionable since its a switch

perhaps "Apply additional compliance controls to this project" should be the label, and the description could expand on what compliance controls are involved